src/server/search/doc_accessors.cc

```diff
@@ -38,13 +38,13 @@ string_view SdsToSafeSv(sds str) {
   return str != nullptr ? string_view{str, sdslen(str)} : ""sv;
 }
 
-search::SortableValue FieldToSortableValue(search::SchemaField::FieldType type, string_view value) {
+search::SortableValue ToSortableValue(search::SchemaField::FieldType type, string_view value) {
   if (type == search::SchemaField::NUMERIC) {
-    double value_as_double = 0;
-    if (!absl::SimpleAtod(value, &value_as_double)) {  // temporary convert to double
+    auto value_as_double = search::ParseNumericField(value);
+    if (!value_as_double) {  // temporary convert to double
       LOG(DFATAL) << "Failed to convert " << value << " to double";
     }
-    return value_as_double;
+    return value_as_double.value();
```

Comment:
this will throw imho if `value_as_double` is nullopt, so it's better use CHECK fail  above.